app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
Ping me on IRC if you need an account, no problem with that.

@AugierLe42e the "comment on my own reshare" is tested in this PR. As you said, it's not the only case.